src/common/node/childProcess.ts

```diff
@@ -38,10 +38,16 @@ interface ISpawnOptions {
 
 export class ChildProcess {
     public static ERROR_TIMEOUT_MILLISECONDS = 300;
+    private childProcess: typeof child_process;
+
+    constructor({childProcess = child_process} = {}) {
```

Comment:
I assume that you've tested this works? My first thought was this might be using javascript's strange declaration semantics where you can do `var x = x;` without complaints, but the variable is undefined because it was assigned to itself.